server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1511,12 +1520,18 @@ public synchronized void updateFromClusterManager(
      * @param primaryTargetAllocationId primary target allocation id
      * @return the replication mode.
      */
-    private boolean isReplicated(String allocationId, String primaryAllocationId, String primaryTargetAllocationId) {
-        // If remote translog is enabled, then returns replication mode checking current allocation id against the
-        // primary and primary target allocation id.
-        // If remote translog is enabled, then returns true if given allocation id matches the primary or it's relocation target allocation
-        // id.
-        if (indexSettings().isRemoteTranslogStoreEnabled()) {
+    private boolean isReplicated(
+        String allocationId,
+        String primaryAllocationId,
+        String primaryTargetAllocationId,
+        boolean assignedToRemoteStoreNode
+    ) {
+        /*
+         - If remote translog is enabled, then returns replication mode checking current allocation id against the primary and primary target allocation id.
+         - If remote translog is enabled, then returns true if given allocation id matches the primary or it's relocation target allocation primary and primary target allocation id.
+         - During an ongoing remote migration, the above-mentioned checks are considered when the shard is assigned to a remote store backed node
+         */
+        if (indexSettings().isRemoteTranslogStoreEnabled() || assignedToRemoteStoreNode == true) {
             return (allocationId.equals(primaryAllocationId) || allocationId.equals(primaryTargetAllocationId));
```

Comment:
can we return false if the if condition holds true since the remote translogs would anyways be downloaded on the relocating primary?